src/main/java/ru/yandex/practicum/filmorate/controllers/ErrorHandler.java

```diff
@@ -0,0 +1,28 @@
+package ru.yandex.practicum.filmorate.controllers;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import ru.yandex.practicum.filmorate.exceptions.SearchedObjectNotFoundException;
+import ru.yandex.practicum.filmorate.exceptions.ValidationException;
+import ru.yandex.practicum.filmorate.model.ErrorResponse;
+
+@Slf4j
+@RestControllerAdvice
+public class ErrorHandler {
+
+    @ExceptionHandler(SearchedObjectNotFoundException.class)
+    @ResponseStatus(HttpStatus.NOT_FOUND)
+    public ErrorResponse objectNotFoundHandler(Exception e) {
+        return new ErrorResponse(e.getMessage(), "404");
```

Comment:
Лучше заменить на уже готовый enum
```suggestion
        return new ErrorResponse(e.getMessage(), String.valueOf(HttpStatus.NOT_FOUND.value()));
```